frontend/src/components/login/Login.tsx

```diff
@@ -37,12 +49,111 @@ export default function LoginComponent() {
+      setErrorMsg("Password should contain 8 characters or more.");
+    } else if (!arePasswordsEqual) {
+      setErrorMsg("Passwords do not match. Please try again.")
+    } else {
+      setErrorMsg("");
+    }
+  }, [password, checkPassword, setPassword, setCheckPassword, arePasswordsEqual]);
+
+  useEffect(() => {
+    if (!(email.includes("@") && email.includes(".")) && email !== "") {
+      setErrorMsg("Email is invalid. Please try again.");
+    } else {
+      setErrorMsg("");
+    }
+  }, [email])
+
+  async function submitNewUser(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    setIsSubmitted(true);
+    let user: User = {
+      name: name,
+      email: email,
+      role: Role.USER,
+    }
+
+    try {      
+      let res = await UserService.createUser(user);
+      router.push(CLIENT_ROUTES.HOME); //TODO: Update with verifying OTP/Email address when auth
+      sessionStorage.setItem("email", res.email);
+    } catch (error) {
+      if (error instanceof PeerPrepErrors.ConflictError) {
+        toast.error("User already exists. Please login instead.", {
+          position: toast.POSITION.BOTTOM_CENTER,
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: false,
+          theme: "dark"
+        })
+      } else {
+        console.log(error);
+        toast.error("Something went wrong. Please refresh and try again.", {
+          position: toast.POSITION.BOTTOM_CENTER,
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: false,
+          theme: "dark"
+        })
+      }
+    } finally {
+      // Cleanup
+      setIsSubmitted(false);
+    }
+  }
+
+  async function getUser(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    try {
+      setIsSubmitted(true);
+      let res = await UserService.getUserByEmail(email);
+      if (res) {
+        sessionStorage.setItem("email", res.email);
+        router.push(CLIENT_ROUTES.HOME);
+      }
+      
+      //TODO: Update with verifying OTP/Email address when auth
+    } catch (error) {
+      if (error instanceof PeerPrepErrors.NotFoundError) {
+        toast.error("User not found. Please sign up instead.", {
+          position: toast.POSITION.BOTTOM_CENTER,
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: false,
+          theme: "dark"
+        })
+      } else {
+        console.log(error);
+        toast.error("Something went wrong. Please refresh and try again.", {
+          position: toast.POSITION.BOTTOM_CENTER,
+          autoClose: 5000,
+          hideProgressBar: true,
+          closeOnClick: true,
+          pauseOnHover: false,
+          draggable: false,
+          theme: "dark"
+        })
```

Comment:
would be good to extract the toast as an independent component for everyone to call, helps to simplify the code stack too 😄